internal/pkg/term/syncbuffer/termprinter.go

```diff
@@ -108,14 +108,16 @@ func (ltp *LabeledTermPrinter) Print() error {
 // If one of the buffer gets done then print entire content of the buffer.
 // Until all the buffers are written to file writer.
 func (ltp *LabeledTermPrinter) printAll() error {
-	for idx, buf := range ltp.buffers {
-		if !buf.IsDone() {
+	for idx := 0; idx < len(ltp.buffers); idx++ {
+		if !ltp.buffers[idx].IsDone() {
 			continue
 		}
 		outputLogs := ltp.buffers[idx].lines()
-		if _, err := ltp.writeLines(buf.label, outputLogs); err != nil {
+		if _, err := ltp.writeLines(ltp.buffers[idx].label, outputLogs); err != nil {
 			return fmt.Errorf("write logs: %w", err)
 		}
+		ltp.buffers = append(ltp.buffers[:idx], ltp.buffers[idx+1:]...)
+		idx--
```

Comment:
huh I've never seen this way of doing deleting from the slice (keeping the index in its place), interesting!

- Can we add a testcase in `TestLabeledTermPrinter_Print` that makes sure that multiple calls to `Print` results in only once printing a buffer to `ltp.term` when it enters `printAll`?